ext/standard/uuid.c

```diff
@@ -0,0 +1,646 @@
+}
+
+PHPAPI zend_always_inline void php_uuid_to_hex(php_uuid_hex *buffer, const php_uuid uuid)
+{
+	sprintf(
+		(char *) buffer,
+		"%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x%02x",
+		uuid[0], uuid[1], uuid[2], uuid[3],
+		uuid[4], uuid[5],
+		uuid[6], uuid[7],
+		uuid[8], uuid[9],
+		uuid[10], uuid[11], uuid[12], uuid[13], uuid[14], uuid[15]
+	);
+}
+
+PHPAPI zend_always_inline void php_uuid_to_string(php_uuid_string *buffer, const php_uuid uuid)
+{
+	sprintf(
+		(char *) buffer,
+		"%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x",
+		uuid[0], uuid[1], uuid[2], uuid[3],
+		uuid[4], uuid[5],
+		uuid[6], uuid[7],
+		uuid[8], uuid[9],
+		uuid[10], uuid[11], uuid[12], uuid[13], uuid[14], uuid[15]
+	);
+}
+
+PHP_MINIT_FUNCTION(uuid)
+{
+	zend_class_entry ce;
+
+	ZEND_BEGIN_ARG_INFO(UUID___construct_args, NULL)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_fromBinary_args, 0, 1, UUID, 0)
+		ZEND_ARG_TYPE_INFO(0, input, IS_STRING, 0)
+		ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_parse_args, 0, 1, UUID, 0)
+		ZEND_ARG_TYPE_INFO(0, input, IS_STRING, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_v3_args, 0, 2, UUID, 0)
+		ZEND_ARG_OBJ_INFO(0, namespace, UUID, 0)
+		ZEND_ARG_TYPE_INFO(0, name, IS_STRING, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_v4_args, UUID, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_v5_args, 0, 2, UUID, 0)
+		ZEND_ARG_OBJ_INFO(0, namespace, UUID, 0)
+		ZEND_ARG_TYPE_INFO(0, name, IS_STRING, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_NamespaceDNS_args, UUID, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_NamespaceOID_args, UUID, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_NamespaceURL_args, UUID, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_NamespaceX500_args, UUID, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_Nil_args, UUID, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_INFO(UUID___clone_args, NULL)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID___set_args, IS_VOID, 0)
+		ZEND_ARG_INFO(0, _)
+		ZEND_ARG_INFO(0, __)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID___wakeup_args, IS_VOID, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID_getVariant_args, IS_LONG, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID_getVersion_args, IS_LONG, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID_isNil_args, _IS_BOOL, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID_toBinary_args, IS_STRING, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID_toHex_args, IS_STRING, 0)
+	ZEND_END_ARG_INFO();
+	ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO(UUID_toString_args, IS_STRING, 0)
+	ZEND_END_ARG_INFO();
```

Comment:
I don't think there's a technical issue with it, assuming PHP only looks for these arginfo structs when the class is declared, because the visibility is `static const` in both cases, so the pointers will work fine.

However, generally PHP style is to keep these declarations outside the MINIT function, keeping code and data separate I guess.